docs/integration/KafkaIntegration.md

```diff
@@ -8,104 +8,164 @@ sidebar_position: 1
+For detailed instruction where it should be placed inside Nussknacker's configuration, take a look at [Configuration details section](#configuration-details)
 
-Configuration necessary to connect with Schema registry should be placed inside `kafkaProperties`. For detailed instruction
-how configuration is handled take a look at [Configuration paragraph](#configuration)
+### Kafka - Connection
 
-#### Confluent-based implementation
+To configure connection to kafka, you need to configure at least `bootstrap.servers` property. It should contain comma separated list of urls to Kafka brokers.
+
+#### Kafka - Authentication
+
+Kafka cluster has multiple options to configure Authentication. Take a look at [Kafka security documentation](https://kafka.apache.org/090/documentation.html#security)
+to see detailed examples how those options should be translated into properties. For example for the typical `SASL_SSL` configuration with
+credential in `JAAS` format, you should provide configuration similar to this one:
+
+```
+kafkaProperties {
+  "schema.registry.url": "http://schemaregistry:8081"
+  "bootstrap.servers": "broker1:9092,broker2:9092"
+  "security.protocol": "SASL_SSL"
+  "sasl.mechanism": "PLAIN"
+  "sasl.jaas.config": "org.apache.kafka.common.security.plain.PlainLoginModule required username=\"some_user\" password=\"some_user\";"
+}
+```
+
+If you use Azure Events Hubs (which uses this mode), username will be `$ConnectionString` and password will be the connection string starting with `Endpoint=sb://`.
+
+In case if you use your own CA and client+server certificates authentication, you should additionally provide:
+`ssl.keystore.location`, `ssl.keystore.password`, `ssl.key.password`, `ssl.truststore.location`, `ssl.truststore.password`.
+
+To make sure if your configuration is correct, you can test it with standard kafka-cli commands
+like `kafka-console-consumer`, `kafka-console-producer` or [kcat](https://github.com/edenhill/kcat).
+
+Some tutorials how to do that:
+- [Kafka quickstart](https://kafka.apache.org/documentation/#quickstart_send)
+- [Azure Event Hubs for Kafka quickstart](https://github.com/Azure/azure-event-hubs-for-kafka/tree/master/quickstart/kafka-cli)
+
+After you'll get properly working set of properties, you just need to copy it to Nussknacker's configuration.
+
+### Schema Registry - Connection
+
+To configure connection Schema Registry, you need to configure at least `schema.registry.url`. It should contain comma separated list of urls to Schema Registry.
+For the single node installation, it will be just an url. Be aware that contrary to Kafka brokers, Schema Registry urls should start with `https://` or `http://`.
+
+Currently, Nussknacker supports two implementations of Schema Registries: based on *Confluent Schema Registry* and based on *Azure Schema Registry*.
+Given implementation is picked based on `schema.registry.url` property. By default, Confluent-based implementation is used.
```

Comment:
I would rephrase it to something like
_Nussknacker determines which registry implementation (Confluent or Azzure) is used from the schema.registry.url property. It the URL ends with .servicebus.windows.net Nussknacker assumes that Azure  schema registry is assumed; if not Confluent schema registry is assumed_.

Reply:
ok